src/framework/tooltip/Tooltip.jsx

```diff
@@ -43,11 +47,46 @@ export default class Tooltip extends Component {
       ) {
         tooltipY = -tooltipDimension.height - 19;
         styles.push('tooltipTop');
+        isPosYSet = true;
       }
 
-      this.container.className = styles.join(' ');
       this.container.style.left = `${tooltipX}px`;
       this.container.style.top = `${tooltipY}px`;
+
+      // reajust tooltip taking into account if it is visible
+      const containerBounds = this.container.getBoundingClientRect();
```

Comment:
I would still consider moving this calculation to a utility method that returns heigh, width, top and left for the position for tooltip. Additionally, may be we can have a few unit tests for that utility method. Depends how easy it is, that would be up to you and @ahollesen if that's feasible or whether makes sense.